python/lsst/ts/eas/eas_csc.py

```diff
@@ -133,27 +137,61 @@ async def handle_summary_state(self) -> None:
         """
         self.log.info(f"handle_summary_state {salobj.State(self.summary_state).name}")
 
-        if self.summary_state == salobj.State.ENABLED:
-            self.enabled_event.set()
-        else:
-            self.enabled_event.clear()
-
         if self.disabled_or_enabled:
-            if self.m1m3_thermal_task.done():
+            # Start the thermal task if not already started
+            if self.m1m3_thermal_task.done() and self.simulation_mode == 0:
                 self.m1m3_thermal_task = asyncio.create_task(self.run_control())
 
+            # Connect if not already connected
             if not self.connected:
                 await self.connect()
         else:
+
+            # Cancel the thermal task
             if not self.m1m3_thermal_task.done():
                 self.m1m3_thermal_task.cancel()
                 try:
                     await self.m1m3_thermal_task
                 except asyncio.CancelledError:
                     pass
 
+            # Disconnect
             await self.disconnect()
 
+        # Set the state of the enabled event
+        if self.summary_state == salobj.State.ENABLED:
+            self.enabled_event.set()
+        else:
+            self.enabled_event.clear()
+
+    async def end_disable(self, id_data: salobj.BaseDdsDataType) -> None:
```

Comment:
I think you might want to do this before the state transition no? 

What this will do is transition the CSC to disable then run the m1m3_stop method, and fault if it fails. I think a better control sequence would be; try to run m1m3_stop then transition to disabled if it works. If it doesn't work, go to fault and raise the exception to fail the command execution.